### PR TITLE
Component eQ footer

### PR DESCRIPTION
**Pull request by armstrongb**
### Context for this PR

Addition of the SDC standard footer component.

- Added the OGL svg icon into assets
- added `_footer2.scss` from the eQ runner repository (https://github.com/ONSdigital/eq-survey-runner/commit/4e9a318e1172a3d32148bf37ed7b32931bb91e77#diff-086b27ddf4ee1c8d30c64fcc323723e7). Some rudimentary tidying of the `_footer2.scss` has been performed but there are elements in there that require closer scrutiny.
- `footer.hbs` html file
- `footer.config.js` for incorporation into the pattern library.
- `readme.md` file for notes on its use.

### How to review this PR

- Checkout the `component-eq-footer` branch and a run yarn start.
- Navigate to the header `component /components/detail/footer` and review each of the component files.

### CODE REVIEW

**Comment by armstrongb** [comment]
Forms part of closing issue #9

File: components/footer/_footer2.scss [hunk 1]
```diff
@@ -0,0 +1,38 @@
+.footer2 {
```
Comment (bennpearson) [inline components/footer/_footer2.scss]: .footer2 needs renaming. Is this replacing the old footer?

File: components/footer/_footer2.scss [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+.footer2_container {
+  width: 1080px;
+  margin: -20px;
+  padding: 0px;
+}
+
+.footer2__link {
+  color: #e4e8eb;
+  &:focus {
+    color: white;
```
Comment (bennpearson) [inline components/footer/_footer2.scss]: Focus colour doesn't have enough contrast. Check here: https://webaim.org/resources/contrastchecker/

File: components/footer/footer.hbs [hunk 3]
```diff
@@ -0,0 +1,56 @@
+<footer class="page__footer">
+  <footer class="footer2 page__footer2">
+    <div class="footer2-nav container" role="contentinfo">
+        <div class="grid container">
+            <div class="">
+                <div class="grid__col col-4@m">
+                    <div class="footer2-nav__heading">
+                        <p class="venus"><strong>Legal information</strong></p>
```
Comment (bennpearson) [inline components/footer/footer.hbs]: Could use heading here

File: assets/img/ogl.svg [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (bennpearson) [inline assets/img/ogl.svg]: Logo isn't very good quality

Comment (bennpearson) [reply assets/img/ogl.svg]: SVG here: https://upload.wikimedia.org/wikipedia/en/4/46/UKOpenGovernmentLicence.svg

File: components/footer/_footer2.scss (hunk 1 above)
Comment (armstrongb) [reply components/footer/_footer2.scss]: This is the service-wide public facing footer, there will eventually be a slimmer version for within the ‘transaction funnel’ of eQ and a different variant for Internal UI.

File: assets/img/ogl.svg (hunk 4 above)
Comment (armstrongb) [reply assets/img/ogl.svg]: Go with the better quality version.

File: components/footer/footer.hbs [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+                        <li class="footer2-nav__item">
+                            <a href="https://www.ons.gov.uk/releasecalendar" target="_blank" class="footer2__link">Release calendar</a>
+                        </li>
+                        <li class="footer2-nav__item">
+                            <a class="footer2__link" href="https://www.ons.gov.uk/news" target="_blank">News</a>
+                        </li>
+                    </ul>
+                    <br>
+                </div>
+                <div class="dl__data mars"><img src="{{path '/assets/img/ogl.svg'}}" class="ogl__img">
```
Comment (bennpearson) [inline components/footer/footer.hbs]: alt attr missing

File: components/footer/_footer.scss [hunk 6]
```diff
@@ -1,13 +1,46 @@
...
+  background-color: $color-very-dark-grey;
+  color: $color-light-grey;
+
+  // ul li { not required as list--bare is available
+  //   list-style-type: none;
+  // }
+  .footer-nav__heading {
+    margin: 2rem 0 1rem;
+  }
+  .dl__data {
```
Comment (hamishtaplin) [inline components/footer/_footer.scss]: This doesn't belong here, it's an element of a `dl` which is nothing to do with the footer. I assume this was used to just use some styling from that element.

File: components/footer/_footer.scss [hunk 7]
```diff
@@ -1,13 +1,46 @@
 .footer {
   padding: 2rem 0;
   margin-top: 2rem;
-  background-color: $color-light-grey;
-  color: $color-very-dark-grey;
+  background-color: $color-very-dark-grey;
+  color: $color-light-grey;
+
+  // ul li { not required as list--bare is available
```
Comment (hamishtaplin) [inline components/footer/_footer.scss]: Remove commented out code. Also, we shouldn't be using element selectors if possible.

File: components/footer/_footer.scss [hunk 8]
```diff
@@ -1,13 +1,46 @@
...
+    margin: 2rem 0 1rem;
+  }
+  .dl__data {
+    padding-left: $grid-gutters;
+  }
+
+}
+
+.footer_container {
+  width: 1080px;
```
Comment (hamishtaplin) [inline components/footer/_footer.scss]: What does this do? It looks like a hack.

File: components/footer/_footer.scss [hunk 9]
```diff
@@ -1,13 +1,46 @@
...
+    outline: 2px dashed #fff;
+    outline-offset: 2px;
+    background: transparent;
+  }
+  &:hover {
+    color: $color-light-grey;
   }
 }
+
+.footer-nav {
```
Comment (hamishtaplin) [inline components/footer/_footer.scss]: Should this be an element of `footer`?

File: components/footer/_footer.scss [hunk 10]
```diff
@@ -1,13 +1,46 @@
...
+  padding: 0 0 2rem;
+}
+// UKOpenGovernmentLicence logo
+.ogl__img {
```
Comment (hamishtaplin) [inline components/footer/_footer.scss]: Where is the rest of this block?

File: components/footer/footer.hbs [hunk 11]
```diff
@@ -0,0 +1,54 @@
+<footer class="page__footer">
+  <footer class="footer page__footer">
+    <div class="footer-nav container" role="contentinfo">
+        <div class="grid container">
+            <div class="grid__col col-4@m">
+                <div class="footer-nav__heading">
+                    <h4 class="venus">Legal information</h4>
+                </div>
+                <ul class="list list--bare footer__list">
```
Comment (hamishtaplin) [inline components/footer/footer.hbs]: `.footer__list` doesn't appear to do anything (no CSS)

File: components/footer/footer.hbs [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+                        <a href="/cookies-privacy" class="footer__link" target="_blank">Cookies and privacy</a>
+                    </li>
+                </ul>
+            </div>
+            <div class="grid__col col-4@m">
+                <div class="footer-nav__heading">
+                    <h4 class="venus">About ONS</h4>
+                </div>
+                <ul class="list list--bare footer__list">
+                    <li class="footer-nav__item">
```
Comment (hamishtaplin) [inline components/footer/footer.hbs]: `.footer-nav__item` has no CSS.

**Comment by armstrongb** [comment]
LGTM @bennpearson can you re-review? Thanks.

File: components/footer/footer.hbs [hunk 13]
```diff
@@ -0,0 +1,64 @@
+<footer class="footer page__footer" role="contentinfo">
```
Comment (hamishtaplin) [inline components/footer/footer.hbs]: Can this be moved to the template of the `page` block?

File: components/footer/README.md [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+### Research
+
+### Usage
+Currently in use in eQ
+
+#### Scope
+Global
+
+#### Dependencies
+* `/components/_footer2.scss`
```
Comment (hamishtaplin) [inline components/footer/README.md]: This needs to be updated

File: components/footer/README.md [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+* `vars/_colours.sccs`
```
Comment (hamishtaplin) [inline components/footer/README.md]: Typo here

File: components/footer/_footer.scss [hunk 16]
```diff
@@ -1,13 +1,40 @@
...
   padding: 2rem 0;
+  padding-bottom : 4rem;
   margin-top: 2rem;
-  background-color: $color-light-grey;
-  color: $color-very-dark-grey;
-}
+  background-color: $color-very-dark-grey;
+  color: $color-light-grey;
+
+  .footer__heading {
```
Comment (hamishtaplin) [inline components/footer/_footer.scss]: Is there any reason to nest these rules? It creates an extra, potentially unnecessary, level of specificity.

File: components/footer/footer.hbs [hunk 17]
```diff
@@ -0,0 +1,64 @@
...
+                        <a href="/cookies-privacy"
+                           class="footer__link"
+                           target="_blank">Cookies and privacy</a>
+                    </li>
+                </ul>
+            </div>
+
+            <div class="grid__col col-4@m">
+                <h4 class="venus footer__heading">About ONS</h4>
+                <ul class="list footer__list">
```
Comment (hamishtaplin) [inline components/footer/footer.hbs]: Is this possibly a variant of a `.nav` or `.list`?

File: components/footer/_footer.scss (hunk 16 above)
Comment (saedwards) [reply components/footer/_footer.scss]: This is much needed I think to help guide ONS developers when following BEM methods. Having seen many examples of invalid BEM usage with elements used outside of blocks (standard BEM allows this). Elements should only exist in the context of a block and this better enforces that rule.

**Pull request merged**